Modules/timemodule.c

```diff
@@ -940,35 +945,50 @@ time_mktime(PyObject *self, PyObject *tup)
 {
     struct tm buf;
     time_t tt;
+#ifdef _AIX
+    time_t clk;
+    int     year = buf.tm_year;
+    int     days = 0;
+#endif
+
     if (!gettmarg(tup, &buf,
                   "iiiiiiiii;mktime(): illegal time tuple argument"))
     {
         return NULL;
     }
-#ifdef _AIX
+#ifndef _AIX
+    buf.tm_wday = -1;  /* sentinel; original value ignored */
+    tt = mktime(&buf);
+#else
     /* year < 1902 or year > 2037 */
-    if (buf.tm_year < 2 || buf.tm_year > 137) {
+    if ((buf.tm_year < 2) || (buf.tm_year > 137))  {
         /* Issue #19748: On AIX, mktime() doesn't report overflow error for
          * timestamp < -2^31 or timestamp > 2**31-1. */
         PyErr_SetString(PyExc_OverflowError,
                         "mktime argument out of range");
         return NULL;
     }
-#else
-    buf.tm_wday = -1;  /* sentinel; original value ignored */
+    /* year < 1970 */
+    if (year < 70) do {
+        buf.tm_year += 4;
+        days -= (366 + (365 * 3));
+    } while (buf.tm_year < 70);
+
+    buf.tm_wday = -1;
+    clk = mktime(&buf);
+    buf.tm_year = year;
+
+    if (buf.tm_wday != -1 && days)
```

Comment:
You still need braces around this if block.

Reply:
you mean:
    if (buf.tm_wday != -1 && days)
        buf.tm_wday = (buf.tm_wday + days) % 7;

should be:
    if (buf.tm_wday != -1 && days) {
        buf.tm_wday = (buf.tm_wday + days) % 7;
    }

?? in other words - bracket a single statement?

Granted - it is confusing which to follow. This code does not bracket single lines:
static PyObject *
time_ctime(PyObject *self, PyObject *args)
{
    time_t tt;
    struct tm buf;
    if (!parse_time_t_args(args, "|O:ctime", &tt))
        return NULL;
    if (_PyTime_localtime(tt, &buf) != 0)
        return NULL;
    return _asctime(&buf);
}

While this block does:
    if (!gettmarg(tup, &buf,
                  "iiiiiiiii;mktime(): illegal time tuple argument"))
    {
        return NULL;
    }

While, for the latter, it could be argued that - as the "if" part is multi-line, using blocks for the "then" part does increase readability.

For now, I am going to pass on the change - but I thank you very much for your critical read. I appreciate it very much!

Reply:
I added additional parentheses - (I see brackets as {})
Thanks!